### PR TITLE
Feature base controller

### PR DESCRIPTION
**Pull request by AlexanderK22**
I've added the BaseController.ts. The App.controller.ts and the Main.controller.ts were also changed to extend from it.

### CODE REVIEW

File: generators/app/templates/src/controller/BaseController.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+/**
+ * @namespace <%= appId %>.controller
+ */
+export default class BaseController extends Controller {
+
+	/**
+	 * Convenience method for getting the view model by name in every controller of the application.
+	 * @public
+	 * @param {string} sName? the model name
```
Comment (vobu) [inline generators/app/templates/src/controller/BaseController.ts:17]: please don‘t type the input params in jsdoc - you casted them already in the code, let TS take care of the rest :)
applies to the other jsdoc blocks as well

File: generators/app/templates/src/controller/BaseController.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	 */
+	public getModel(sName?: string) : Model {
+		return this.getView().getModel(sName);
+	}
+
+	/**
+	 * Convenience method for setting the view model in every controller of the application.
+	 * @public
+	 * @param {sap.ui.model.Model} oModel the model instance
+	 * @param {string} sName? the model name
```
Comment (vobu) [inline generators/app/templates/src/controller/BaseController.ts:28]: optional params should be denoted like [sName]

**Review by vobu — CHANGES_REQUESTED** [review]
only small changes in jsdoc, otherwise good to go!

**Comment by AlexanderK22** [comment]
I've updated the JSDoc in the BaseController.js

File: generators/app/templates/src/controller/Main.controller.ts [hunk 3]
```diff
@@ -1,13 +1,13 @@
 import MessageBox from "sap/m/MessageBox";
-import Controller from "sap/ui/core/mvc/Controller";
+import BaseController from "<%= appURI %>/controller/BaseController";
```
Comment (petermuessig) [inline generators/app/templates/src/controller/Main.controller.ts:2]: Same here, relative URL should be sufficient.

**Review by petermuessig — COMMENTED** [review]
Sorry, for being late to the party. I like your suggestion and have just two minor comments on that - but we can even submit it without changing.

**Comment by AlexanderK22** [comment]
@petermuessig Using a relative URL is probably the better option. I've updated it in the App and Main controller. Thanks for the suggestion 👍.

**Review by petermuessig — CHANGES_REQUESTED** [review]
Hi @AlexanderK22 ,

sorry to bother you a bit more. I created the application now and run the eslint checks `yarn lint` and the following issues appear:

image

Would you be able to catch them? Or should I try to assist here? 

I think, in future, we need to add to run the lint on the generated application to validate the content upfront - overall to introduce some tests.

THX and BR,
Peter

**Comment by AlexanderK22** [comment]
Hi @petermuessig,
I'd appreciate your assitance in this matter, to resolve the issue as quickly as possible.

Thanks in advance.

**Comment by petermuessig** [comment]
@AlexanderK22 - ok, I'll try to reserve some time today evening or tomorrow morning to fully sort this out. Big thanks to your contribution so far... 👍

**Comment by AlexanderK22** [comment]
@petermuessig thanks again. Glad I could help out a bit.

**Comment by petermuessig** [comment]
@AlexanderK22 @vobu - codebase is cleaned-up now and should be good to go. The eslint checks are now reporting no issue anymore. 😄

**Comment by petermuessig** [comment]
@AlexanderK22 - thanks for your contribution. It's merged now, feel free to test it and provide feedback... 😄

**Comment by AlexanderK22** [comment]
@petermuessig Great! I will test it and let you know if there are any issues.

**Pull request merged**